### ISSUE TITLE
disable InexactError for runtime parsing

### ISSUE THREAD

**Issue by stevengj** [JuliaMath/DecFP.jl#127]
Currently, 
```jl
julia> parse(Dec32, "1.999999999999999999")
ERROR: InexactError: parse(Dec32, 1.999999999999999999)

julia> parse(Float32, "1.999999999999999999")
2.0f0
```
which is rather inconsistent.   My feeling is that we should disable `InexactError` for runtime calls to `parse` and just return the closest rounded value.

On the other hand, we should probably *keep* the error for compile-time constants via the `dXX"..."` string macros
```jl
julia> d32"1.999999999999999999"
ERROR: LoadError: InexactError: parse(Dec32, 1.999999999999999999)
```
since in these cases it seems likely that an exact decimal constant was intended and an error is more useful than a rounded result.

**Comment by jmkuhn** [comment]
Yes, I have been looking at this as well.

**Comment by jmkuhn** [comment]
Currently on master I have
```
julia> parse(Dec32, "1.999999999999999999")
2.0

julia> parse(Float32, "1.999999999999999999")
2.0f0

julia> d32"1.999999999999999999"
2.0
```
which is what we want for `parse` but not for the string macros.  Changing this would be breaking so this should be resolved before 1.0.

**Comment by stevengj** [comment]
Resolved.
